applications/FSIApplication/custom_python/add_convergence_accelerators_to_python.cpp

```diff
@@ -38,52 +38,50 @@ void AddConvergenceAcceleratorsToPython(pybind11::module &m)
+    typedef typename MVQNFullJacobianConvergenceAcceleratorType::Pointer MVQNFullJacobianConvergenceAcceleratorPointerType;
+    py::class_<MVQNFullJacobianConvergenceAcceleratorType, MVQNFullJacobianConvergenceAcceleratorPointerType, BaseConvergenceAcceleratorType>(m, "MVQNFullJacobianConvergenceAccelerator")
+        .def(py::init<Parameters>())
         .def(py::init<double, double>())
-        .def("InitializeSolutionStep", &MVQNFullJacobianConvergenceAccelerator<TSpace>::InitializeSolutionStep)
-        .def("UpdateSolution", &MVQNFullJacobianConvergenceAccelerator<TSpace>::UpdateSolution)
-        .def("FinalizeNonLinearIteration", &MVQNFullJacobianConvergenceAccelerator<TSpace>::FinalizeNonLinearIteration)
-        .def("FinalizeSolutionStep", &MVQNFullJacobianConvergenceAccelerator<TSpace>::FinalizeSolutionStep);
+        .def("InitializeSolutionStep", &MVQNFullJacobianConvergenceAcceleratorType::InitializeSolutionStep)
+        .def("UpdateSolution", &MVQNFullJacobianConvergenceAcceleratorType::UpdateSolution)
+        .def("FinalizeNonLinearIteration", &MVQNFullJacobianConvergenceAcceleratorType::FinalizeNonLinearIteration)
+        .def("FinalizeSolutionStep", &MVQNFullJacobianConvergenceAcceleratorType::FinalizeSolutionStep);
 
     // MVQN recursive convergence accelerator
-    py::class_<MVQNRecursiveJacobianConvergenceAccelerator<TSpace>, BaseConvergenceAcceleratorType>(m, "MVQNRecursiveJacobianConvergenceAccelerator")
-        .def(py::init<Parameters &>())
+    typedef MVQNRecursiveJacobianConvergenceAccelerator<SparseSpaceType, DenseSpaceType> MVQNRecursiveJacobianConvergenceAcceleratorType;
+    typedef typename MVQNRecursiveJacobianConvergenceAcceleratorType::Pointer MVQNRecursiveJacobianConvergenceAcceleratorPointerType;
+    py::class_<MVQNRecursiveJacobianConvergenceAcceleratorType, MVQNRecursiveJacobianConvergenceAcceleratorPointerType, BaseConvergenceAcceleratorType>(m, "MVQNRecursiveJacobianConvergenceAccelerator")
+        .def(py::init<Parameters>())
         .def(py::init<double, unsigned int, double>())
-        .def("Initialize", &MVQNRecursiveJacobianConvergenceAccelerator<TSpace>::Initialize)
-        .def("InitializeSolutionStep", &MVQNRecursiveJacobianConvergenceAccelerator<TSpace>::InitializeSolutionStep)
-        .def("UpdateSolution", &MVQNRecursiveJacobianConvergenceAccelerator<TSpace>::UpdateSolution)
-        .def("FinalizeNonLinearIteration", &MVQNRecursiveJacobianConvergenceAccelerator<TSpace>::FinalizeNonLinearIteration);
+        .def("Initialize", &MVQNRecursiveJacobianConvergenceAcceleratorType::Initialize)
+        .def("InitializeSolutionStep", &MVQNRecursiveJacobianConvergenceAcceleratorType::InitializeSolutionStep)
+        .def("UpdateSolution", &MVQNRecursiveJacobianConvergenceAcceleratorType::UpdateSolution)
+        .def("FinalizeNonLinearIteration", &MVQNRecursiveJacobianConvergenceAcceleratorType::FinalizeNonLinearIteration);
```

Comment:
except the constructors you can remove all exposures of the other functions, this is now done through the baseclass

Reply:
You're right. Copy paste error.